joerd/check.py

```diff
@@ -19,6 +20,21 @@ def is_zip(tmp):
     return False
 
 
+def is_tar_gz(tmp):
+    """
+    Returns True if the NamedTemporaryFile given as the argument appears to be
+    a GZip-encoded TAR file.
+    """
+
+    try:
+        tar = tarfile.open(tmp.name, mode='r:gz', errorlevel=2)
+        names = [info.name for info in tar]
```

Comment:
I changed the check in 74fd81e to closely resemble what the rest of the code actually does with the data. It should catch any errors that would happen later - although it does now try to decompress the file. Compared to the network overhead of downloading the file, it should be relatively cheap though.

Does that look readable & understandable?